node/core/approval-voting/src/approval_db/v1/mod.rs

```diff
@@ -161,11 +164,16 @@ pub struct Config {
 /// Details pertaining to our assignment on a block.
 #[derive(Encode, Decode, Debug, Clone, PartialEq)]
 pub struct OurAssignment {
-	pub cert: AssignmentCert,
+	/// Our assignment certificate.
+	pub cert: AssignmentCertV2,
+	/// The tranche for which the assignment refers to.
 	pub tranche: DelayTranche,
+	/// Our validator index for the session in which the candidates were included.
 	pub validator_index: ValidatorIndex,
-	// Whether the assignment has been triggered already.
+	/// Whether the assignment has been triggered already.
 	pub triggered: bool,
+	/// A subset of the core indices obtained from the VRF output.
+	pub assignment_bitfield: CoreBitfield,
```

Comment:
I don't follow why this is in the `OurAssignment`. The `OurAssignment` is stored in the approval entry (i.e. a particular candidate in a particular block). This bitfield refers to other candidates included in the same block - doesn't that mean it's being duplicated way more than necessary?

Reply:
Yes it is duplicated, but it makes it easier to distribute the assignment which now can refer to multiple candidates. I could wrap in an `Arc` to have a single instance of it.

Reply:
This is related to a todo in the code `// TODO: Fix warning: `Importing locally an already known assignment` for multiple candidate assignments.` . I haven't yet figured out how to solve that properly

Reply:
It'd hit the DB in any case; most of this stuff is probably in the in-memory cache of rocksdb and is repeatedly serialized. Not a blocker but feels funny to duplicate it so much.

Reply:
Yeah it is funny, but didn’t fix it because I couldn’t find an easy way yet. I am still sleeping on it.

Reply:
This is now fixed, `assignment_bitfield` is gone.